codec.go

```diff
@@ -308,6 +308,30 @@ func (cdc *Codec) PrintTypes(out io.Writer) error {
 	return nil
 }
 
+// ConcreteRegisteredName returns the name under which i has been registered.
+func (cdc *Codec) ConcreteRegisteredName(i interface{}) (string, error) {
+	cdc.mtx.RLock()
+	defer cdc.mtx.RUnlock()
+
+	if i == nil {
+		return "", errors.New("argument must not be nil")
+	}
+
+	gt := reflect.TypeOf(i)
+
+	// if i is a non-nil pointer, dereference it and grab its inner Elem
+	if gt.Kind() == reflect.Ptr {
+		gt = gt.Elem()
+	}
+
+	ct, ok := cdc.typeInfos[gt]
+	if !ok {
+		return "", fmt.Errorf("no type registered for %s", gt.String())
```

Comment:
This should probably be a sentinel error, such that callers can switch over a returned error to see if the type wasn't registered or if any other error occurred.

And independent of that: I think `%s` already calls `String()` on the type (no need to explicitly call it here).

Reply:
Would those two errors be okay for the purpose?
 - `ErrNilPointer`, when `i` is `nil`;
 - `ErrTypeNotRegistered`, when the concrete type of `i` is not registered;

Yup, `String()` will be removed :-)